src/main/kotlin/com/falcon/falcon/entrypoint/rest/user/UserRequest.kt

```diff
@@ -3,18 +3,18 @@ package com.falcon.falcon.entrypoint.rest.user
 import com.falcon.falcon.core.entity.User
 import javax.validation.constraints.NotBlank
 import org.hibernate.validator.constraints.Length
-import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
 
 data class UserRequest(
     @field:NotBlank(message = "username must be sent")
-    val username: String,
+    val username: String?,
+    @field:NotBlank(message = "password must be sent")
     @field:Length(min = 3, max = 64, message = "password must be between 3 and 64 characters")
-    val password: String
+    val password: String?
 )
 
-fun UserRequest.toDomain(bCrypt: BCryptPasswordEncoder) =
+fun UserRequest.toDomain() =
     User(
-        username = this.username,
-        password = bCrypt.encode(this.password)
+        username = this.username!!,
+        password = this.password!!
```

Comment:
Será q se não o @valid suporta validar a falta de um campo no json ?

Ex: caso eu mande um json com dois valores e eles tiverem a validação o @valid irá agir mas caso eu não mande esse campo ele iria entrar como null e a anotação irá pegar esse campo tbm ?

Não lembro se temos uma validação para ver se não e null, caso não tenhamos penso q podemos tomar um null pointer durante a validação da anotação do filed pois o spring irá considerar que o campo não e null e irá aplicação a validação do not blank.

Uma ideia a se pensar

Reply:
acho que eu ate cheguei a falar sobre isso na call, mas se o campo não for nullable, a validacao do @notblank (que também valida not null) não funciona já que o campo nunca pode ser nulo, nesse caso ele roda uma outra exceção propria do kotlin, por isso esse campo precisa ser nullable
caso ele seja null, ele cai na validação e é retornado a exception certa.

Reply:
Ahhh vdd man